libopenage/gui/guisys/private/opengl_debug_logger.cpp

```diff
@@ -0,0 +1,131 @@
+// Copyright 2017-2017 the openage authors. See copying.md for legal info.
+
+#include "opengl_debug_logger.h"
+
+#include <cassert>
+
+#include <QOpenGLDebugLogger>
+
+namespace qtsdl {
+GLDebugParameters get_current_opengl_debug_parameters(const QOpenGLContext &current_source_context)  {
+	GLDebugParameters params;
+
+#ifndef __APPLE__
+	if ((params.is_debug = current_source_context.format().options().testFlag(QSurfaceFormat::DebugContext)))
+	{
+		glGetPointerv(GL_DEBUG_CALLBACK_FUNCTION, &params.callback);
+		params.synchronous = glIsEnabled(GL_DEBUG_OUTPUT_SYNCHRONOUS);
+	}
+#endif
+
+	return params;
+}
+
+std::unique_ptr<QOpenGLDebugLogger> logger_from_opengl_debug_parameters(GLDebugParameters params, QOpenGLContext &current_dest_context)  {
+	assert(QOpenGLContext::currentContext() == &current_dest_context);
+
+	if (params.is_debug) {
+		auto logger = std::make_unique<QOpenGLDebugLogger>();
+		logger->initialize();
+
+#ifndef __APPLE__
+		if (params.callback) {
+			auto cb = reinterpret_cast<void (*)(GLenum, GLenum, GLuint, GLenum, GLsizei, const GLchar*, void*)>(params.callback);
+			QObject::connect(&*logger, &QOpenGLDebugLogger::messageLogged, [cb] (const QOpenGLDebugMessage &debugMessage) {
+				GLenum source;
+				GLenum type;
+				GLenum severity;
+
+				switch (debugMessage.source()) {
+				case QOpenGLDebugMessage::APISource:
+					source = GL_DEBUG_SOURCE_API;
+					break;
```

Comment:
we already have that in `libopenage/error/gl_debug.cpp`, we should not duplicate it. any idea how we could unify them?